homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+"""Support for Watergate sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import UTC, datetime
+import logging
+
+from watergate_local_api.models import AutoShutOffReport
+
+from homeassistant.components.sensor import (
+    HomeAssistant,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolume,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WatergateConfigEntry
+from .coordinator import WatergateAgregatedRequests, WatergateDataCoordinator
+from .entity import WatergateEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(kw_only=True, frozen=True)
+class WatergateSensorEntityDescription(SensorEntityDescription):
+    """Description for Watergate sensor entities."""
+
+    value_fn: Callable[[WatergateAgregatedRequests], str | int | float | None]
+
+
+DESCRIPTIONS: list[WatergateSensorEntityDescription] = [
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_volume",
+        key="water_meter_volume",
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_duration",
+        key="water_meter_duration",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ip if data.networking else None,
+        translation_key="ip",
+        key="ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.gateway if data.networking else None,
+        translation_key="gateway",
```

Comment:
I don't like having this as a sensor. Instead imo it would make more sense to add the gateway as it's own device and set `via_device` in the `DeviceInfo`

Reply:
It is a network gateway of this device - not a separate device.

Reply:
then I don't know what this tells me actually

Reply:
It is network configuration like device IP address, subnet and other stuff. I think it is valuable data sometimes, at least for me. This is why it is Diagnostic.

Reply:
If someone would require this data I would suggest just adding it to a diagnostics.py. These sensors have little to no automation value. The IP address can be useful, but that something we should solve globally in Home Assistant and not per integration

Reply:
@joostlek So what you are saying that I should remove those sensors from this PR and then in another one introduce diagnostics, right?

Reply:
I think that would work the best yes

Reply:
@joostlek So what you are saying that I should remove those sensors from this PR and then in another one introduce diagnostics, right?

Reply:
Just noted that Matter devices share their IP etc like this:

<img width="360" alt="Bildschirmfoto 2024-12-18 um 13 11 39" src="https://github.com/user-attachments/assets/5ce76378-3155-4c2e-83cd-886941387251" />

Could that be a solution for this device as well?